tests/bracket.test

```diff
@@ -27,3 +27,16 @@ incAll ys x = map (+x) fs
 
 OUTPUT
 No hints
+
+---------------------------------------------------------------------
+RUN tests/bracket-constraints.hs
+FILE tests/bracket-constraints.hs
+
+foo :: (A) => ()
```

Comment:
These tests can be written directly in Bracket.hs. The `tests` dir contains tests that can't be written in the source files (e.g., because they need to read config from `.hlint.yaml`).